### DISCUSSION TITLE
Running migration script file with prisma migrate command

### DISCUSSION THREAD

**Discussion by leungkinghin-ct** [prisma/prisma#25016] [Q&A] [answered]
### Question

I read this [documentation](https://www.prisma.io/docs/orm/prisma-migrate/workflows/data-migration#create-and-run-the-data-migration-file) which is about data migration with script file.  I followed the guideline and put the javascript migration script file inside generated folder in `prisma/migration`. However the documentation only provides guideline how we can execute the migration file with npm command defined in `package.json` as below.  

```
"scripts": {
    "dev": "ts-node ./script.ts",
    "data-migration:add-status-column": "ts-node ./prisma/migrations/20230417131956_add-status-column/data-migration.ts"
},
```

May i know whether we can execute the script file with `prisma migrate` command? so that i don't need to specify the script file path in `package.json`.

Thanks.

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by nurul3101** [comment]
Hey @leungkinghin-ct 👋 

No, `prisma migrate` command cannot execute the TypeScript file directly. So, for data migrations you should go with the suggested approach of specifying the script name in your `package.json` file. 

Just a note that it's not necessary to specify the script name in the `package.json` file. If this is a one time data migration, then you can directly execute the TypeScript file through your CLI without specifying the script name.

You can run `ts-node ./prisma/migrations/20230417131956_add-status-column/data-migration.ts` file through your CLI and it should run fine.

**Reply by leungkinghin-ct** [reply]
Thanks @nurul3101 

That means the typescript file needs not to be put under the prisma generated folder, is it right? As long as the typescript file can be accessed, it can be still executed through CLI no matter where the typescript file is placed.

**Reply by nurul3101** [reply, marked as answer]
Yes, that's correct.

**Reply by leungkinghin-ct** [reply]
Got it. Thank you for your feedback

**Comment by michaelstone** [comment]
We've just encountered this issue too. Running the data migration through the CLI is a really awkward way of doing this for an automated deployment, e.g. tracking whether the data migration has previously been run.

Are there any plans to allow data migrations to be run as part of prisma deploy?

**Comment by GuysmoB** [comment]
Hi,

I don't understand how I'm suppose to run the npm command.
I'm using  Vercel but not the CLI. 
If I add a new script in my `package.json,` it will run for every deployment ?

```
"scripts": {
       "dev": "dotenv -e .env.local -- next dev",
	"build": "dotenv -e .env.prod -- next build",
	"start": "dotenv -e .env.prod -- next start",
	"lint": "next lint",
	"postinstall": "prisma generate && prisma migrate deploy"
},
```

Thanks for any help !

**Reply by mkschreder** [reply]
No it will not run automatically. This is indeed very awkward and error probe when data migration has to be done as a separate step.
